editor/editor_file_dialog.cpp

```diff
@@ -760,8 +760,13 @@ void EditorFileDialog::update_file_list() {
 
 	while ((item = dir_access->get_next()) != "") {
 
-		if (item == "." || item == "..")
+		if (item == "." || item == ".." || item == ".gdignore") {
```

Comment:
Does this actually work?

`.gdignore` is a file that should be placed in folders that are meant to be ignored.
Here you're only ignoring the file itself, no?

Reply:
Hmmm... It can also be a directory, so I cover this case.

Ignoring a directory with `.gdignore` file would be more complex.

Reply:
Well the `.gdignore` feature is meant for a file in a directory, not a directory itself. There's little use for a hidden `.gdignore` directory in a Godot project IMO, but there's use for putting a `.gdignore` file in your `docs/` folder to avoid importing your design document images as resources.

Reply:
> Well the .gdignore feature is meant for a file in a directory, not a directory itself.

But it works for a directory too (i.e. any `.gdignore` directory is ignored by the editor).

Reply:
Well the editor ignores all hidden files, no?

Reply:
I'm on Windows, so I didn't even notice. But it looks like it has the same problem - only File System dock ignores the hidden files, the file dialog shows them.

Maybe we should just ignore all files/directories starting with dot. This would handle `.import` too 🤔

Reply:
Well that's what `show_hidden_files` does just below.

Reply:
Ok, so I just removed this check. This PR was only about `.import` anyways, `.gdignore` is another issue.